data/actions/scripts/other/potions.lua

```diff
@@ -1,158 +1,230 @@
-			player:say("This potion can only be consumed by knights of level 80 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
+		if potion.manaToAdd then
+			local min, max = potion.manaToAdd.min * (potions.manaRate or 1), potion.manaToAdd.max * (potions.manaRate or 1)
+			if not doTargetCombatMana(0, target, min, max, effect) then
+				return false
+			end
 		end
 
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 425, 575, CONST_ME_MAGIC_BLUE) then
-			return false
+		if potion.combat then
+			if not potion.combat:execute(target, numberToVariant(target:getId())) then
+				return false
+			end
 		end
 
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(greatEmptyPot, 1)
-	elseif itemId == greatManaPot then
-		if (not isInArray({1,2,5,6}, target:getVocation():getId()) or target:getLevel() < 80) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by sorcerers and druids of level 80 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
+		if potion.condition and potion.condition:getType() ~= CONDITION_EXHAUST_HEAL then
+			if not player:addCondition(potion.condition) then
+				return false
+			end
 		end
 
-		if not doTargetCombatMana(0, target, 150, 250, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(greatEmptyPot, 1)
-	elseif itemId == ultimateHealthPot then
-		if (not isInArray({4, 8}, target:getVocation():getId()) or target:getLevel() < 130) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by knights of level 130 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
+		if not potions.words then
+			potions.words = "Aaaah..."
 		end
 
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 650, 850, CONST_ME_MAGIC_BLUE) then
-			return false
+		target:say(potions.words, TALKTYPE_MONSTER_SAY)
+
+		-- The default behavior is removing even if nil.
+		if potions.removePotions ~= false then
+			if potions.addEmptyFlasks ~= false and potion.emptyFlask then
+				-- potion's empty potion should be in player's backpack only if the potion was in his backpack!
+				if item:getTopParent().uid == player.uid then
+					player:addItem(potion.emptyFlask)
+				else
+					Game.createItem(potion.emptyFlask, 1, item:getPosition())
+				end
+			end
+
+			-- Removing potion at the end to avoid <nil> error.
+			item:remove(1)
 		end
 
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(greatEmptyPot, 1)
+		if not getPlayerFlagValue(player, PlayerFlag_HasNoExhaustion) then
+			player:addCondition(potions.exhaust) -- Last thing to do is adding condition!
+		end
+		return true
 	end
-	return true
+	return false -- you can't use this object (potion isn't added to the table.)
```

Comment:
I don't think it will ever be the case since it's only triggered for items listed in the XML and the user has to be über dumb to edit there and don't edit here or vice-versa.